### PR TITLE
Closes #10844: Only load app links URL if scheme is supported by the engine

### LINKED ISSUE #10844: Only load app links URL if scheme is supported by the engine.
Currently, when user decides not to launch app links in private tab, we try to load the URL.  However, there are cases when the URL is not supported by the engine.  In these cases we should not try to load it.

This is for https://github.com/mozilla-mobile/focus-android/issues/5183

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
Tested with both Fenix and Focus.

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksFeature.kt [hunk 1]
```diff
@@ -52,7 +54,8 @@ class AppLinksFeature(
     private val launchInApp: () -> Boolean = { false },
     private val useCases: AppLinksUseCases = AppLinksUseCases(context, launchInApp),
     private val failedToLaunchAction: () -> Unit = {},
-    private val loadUrlUseCase: SessionUseCases.DefaultLoadUrlUseCase? = null
+    private val loadUrlUseCase: SessionUseCases.DefaultLoadUrlUseCase? = null,
```
Comment (eliserichards) [inline components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksFeature.kt:57]: Why does this have a `null` default? would there ever be a case where the scheme is supported but we don't have this use case?

**Review by eliserichards — APPROVED** [review]
👍 🚀

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksFeature.kt (hunk 1 above)
Comment (rocketsroger) [reply components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksFeature.kt:57]: There could be a case that the user of `AppLinksFeature` decides not to use the LoadUrlUseCase.  In that case they can pass in a null and we won't try to load URLs through `AppLinksFeature`.

**Comment by rocketsroger** [comment]
@Mergifyio backport releases/91.0

**Comment by rocketsroger** [comment]
@Mergifyio backport releases/92.0

**Pull request merged**